api/xds_protocol.rst

```diff
@@ -155,45 +258,110 @@ the resources in the :ref:`DiscoveryResponse <envoy_api_msg_DiscoveryResponse>`
-An empty EDS/RDS :ref:`DiscoveryResponse <envoy_api_msg_DiscoveryResponse>` is effectively a nop from the perspective of the
-respective resources in the Envoy.
-
-When a `Listener` or `Cluster` is deleted, its corresponding EDS and
-RDS resources are also deleted inside the Envoy instance. In order for
-EDS resources to be known or tracked by Envoy, there must exist an
-applied `Cluster` definition (e.g. sourced via CDS). A similar
-relationship exists between RDS and `Listeners` (e.g. sourced via
-LDS).
+the *resource_names* list, the server must resend any newly requested resources, even if it previously sent those resources
+without having been asked for them and the resources have not changed since that time.
+
+Grouping Resources into Responses
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+In the SotW API variants, the `Listener` and `Cluster` resource types are treated specially. The server must include the complete
+state of the world, meaning that all resources of the relevant type that are needed by the client must be included, even if they
+did not change since the last response. This means that if the server has previously sent 100 resources and only one of them has
+changed, it must resend all 100 of them, even the 99 that were not modified. However, all other resource types may be handled
+incrementally in the SotW API, meaning that resources will be grouped into responses the same way as they would be in the
+incremental API.
+
+In the incremental API variants, the server sends each resource in its own response. This means that if the server has previously
+sent 100 resources and only one of them has changed, it may send a response containing only the changed resource; it does not
+need to resend the 99 resources that have not changed, and the client must not delete the unchanged resources.
+
+Deleting Resources
+^^^^^^^^^^^^^^^^^^
+
+In the incremental API variants, the server signals the client that a resource should be deleted via the :ref:`removed_resources
+<envoy_api_field_DeltaDiscoveryResponse.removed_resources>` field of the response. This tells the client to remove the
+resource from its local cache.
+
+In the SotW API variants, the criteria for deleting resources is more complex. For `Listener` and `Cluster` resource types,
+if a previously seen resource is not present in a new response, that indicates that the resource has been removed, and the
+client must delete it; a response containing no resources means to delete all resources of that type. However, for other
+resource types, the API provides no mechanism for the server to tell the client that resources have been deleted; instead,
+deletions are indicated implicitly by parent resources being changed to no longer refer to a child resource. (For example,
+when the client receives an LDS update removing a Listener that was previously pointing to RouteConfiguration A, if no other
```

Comment:
Nit: no need for parens around the `For example`, also, we should use API type markup as described above here.

Reply:
Done.